core/frontend/src/components/common/DevicePathHelper.vue

```diff
@@ -0,0 +1,191 @@
+      <v-icon
+        v-if="board_connector"
+        class="ml-4"
+        v-bind="attrs"
+        v-on="on"
+      >
+        mdi-information
+      </v-icon>
+    </template>
+    <img
+      :src="board_image"
+      style="max-height:500px;"
+    >
+    <div :class="circle_class" />
+  </v-tooltip>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+
+import system_information from '@/store/system-information'
+import { Dictionary } from '@/types/common'
+import { SerialPortInfo } from '@/types/system-information/serial'
+
+enum BoardType {
+  Rpi4B = 'Rpi4B',
+  Rpi3B = 'Rpi3B',
+  Navigator = 'Navigator',
+  Unknown = 'Unknown'
+}
+
+const connector_map: Dictionary<string> = {
+  '/dev/ttyS0': 'serial1',
+  '/dev/ttyAMA1': 'serial3',
+  '/dev/ttyAMA2': 'serial4',
+  '/dev/ttyAMA3': 'serial5',
+  '/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.3:1.0-port0': 'usb-top-left-pi4',
+  '/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.4:1.0-port0': 'usb-bottom-left-pi4',
+  '/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.1:1.0-port0': 'usb-top-right-pi4',
+  '/dev/serial/by-path/platform-fd500000.pcie-pci-0000:01:00.0-usb-0:1.2:1.0-port0': 'usb-bottom-right-pi4',
+  '/dev/serial/by-path/platform-3f980000.usb-usb-0:1.5:1.0-port0': 'usb-bottom-right-pi3',
+  '/dev/serial/by-path/platform-3f980000.usb-usb-0:1.4:1.0-port0': 'usb-top-right-pi3',
+  '/dev/serial/by-path/platform-3f980000.usb-usb-0:1.3:1.0-port0': 'usb-bottom-left-pi3',
+  '/dev/serial/by-path/platform-3f980000.usb-usb-0:1.2:1.0-port0': 'usb-top-left-pi3',
+}
+
+export default Vue.extend({
+  name: 'DevicePathHelper',
+  props: {
+    device: {
+      type: String,
+      required: true,
+    },
+  },
+  computed: {
+    serial_port_path(): string {
+      /* returns the by-path path for the serial port if available */
+      const system_serial_ports: SerialPortInfo[] | undefined = system_information.serial?.ports
+      if (system_serial_ports === undefined || system_serial_ports.isEmpty()) {
+        return this.device as string
+      }
+      const port = system_serial_ports.find((a) => a.name === this.device)
+      return port?.by_path ?? this.device as string
+    },
+    board_type() : BoardType {
+      /* Detects board type between navigator and Rpi4 */
+      // TODO: make sure this works with pi3 as well
+      const serial_port_path = this.serial_port_path as string // why is it needed?
+      if (serial_port_path.includes('ttyAMA') || serial_port_path.includes('ttyS0')) {
+        return BoardType.Navigator
+      }
+      if (serial_port_path.includes('platform-3f980000')) {
+        return BoardType.Rpi3B
+      }
+      if (serial_port_path.includes('platform-fd500000')) {
+        return BoardType.Rpi4B
+      }
+      return BoardType.Unknown
+    },
+    board_image() : string {
+      switch (this.board_type) {
+        case BoardType.Navigator:
+          /* eslint-disable */
```

Comment:
It would be nice to know what eslint-disable is disabling, please specify the linter feature that you want to disable and not everything
```
// eslint-disable-next-line @typescript...
```